compiler/src/iree/compiler/Codegen/Common/OptimizeTensorInsertExtractSlices.cpp

```diff
@@ -253,10 +253,34 @@ struct CastLikeInsertSliceOpFolder final
 };
 } // namespace
 
+// Find the latest insertion point in the block for the given values.
```

Comment:
I dont quite understand the method taking `values`... you want latest insertion point for an operation. 
Also better to use dominance info (which anyway uses the `isBeforeInBlock`).

Nit: This doesnt read like "latest" insertion point. It is the earliest insertion point. You can insert it later than this method is finding. But if you insert it earlier that would be wrong.